src/qml/components/BlockClock.qml

```diff
@@ -42,7 +42,8 @@ Item {
         id: dial
         anchors.horizontalCenter: root.horizontalCenter
         scale: Theme.blockclocksize
-        width: Math.min((root.parentWidth * dial.scale), (root.parentHeight * dial.scale))
+        width: {Math.max(Math.min(200, Math.min(root.parentWidth - 30, root.parentHeight - 30)), 
+                Math.min((root.parentWidth * dial.scale), (root.parentHeight * dial.scale)))}
```

Comment:
nit: I would extract the formula to a function where I'd structure the logic as @GBKS suggested [above](https://github.com/bitcoin-core/gui-qml/pull/385#issuecomment-1865777711).

<details>
<summary>How about this...</summary>**

```diff --git a/src/qml/components/BlockClock.qml b/src/qml/components/BlockClock.qml
--- a/src/qml/components/BlockClock.qml
+++ b/src/qml/components/BlockClock.qml
@@ -44,7 +44,7 @@ Item {
         id: dial
         anchors.horizontalCenter: root.horizontalCenter
         scale: Theme.blockclocksize
-        width: Math.min((root.parentWidth * dial.scale), (root.parentHeight * dial.scale))
+        width: calculateConstrainedWidth()
         height: dial.width
         penWidth: dial.width / 50
         timeRatioList: chainModel.timeRatioList
@@ -239,6 +239,24 @@ Item {
         }
     ]

+    function calculateConstrainedWidth() {
+        // Margin size used to create a buffer around the constrained area.
+        var marginSize = 30;
+
+        // The maximum size.
+        // This is our hardest constraint - the physical screen size.
+        var screenConstraintSize = Math.min(root.parentWidth, root.parentHeight) - marginSize;
+
+        // The minimum size (for legibility).
+        var minimumSize = 200;
+
+        // The dynamic target size based on our display mode scale.
+        // Basically just scaling up for larger non-mobile screens.
+        var scaleTargetSize = Math.min(root.parentWidth, root.parentHeight) * dial.scale;
+
+        // Mashing them all up
+        return Math.min(screenConstraintSize, Math.max(minimumSize, scaleTargetSize))
+    }

     function formatProgressPercentage(progress) {
         if (progress >= 1) {
```
</details>

Reply:
thanks for testing on a wearable device,  maybe let’s open a new PR if this feels like a priority, what do you think @GBKS?

Reply:
I mocked things up and added them to the design docs, [see here](https://bitcoincore.app/block-clock/#layout-logic). Feedback appreciated.

Reply:
Also had a question about how this works on Android Wear. Is the full app running on the watch? Or does it run in some sort of stripped down way? I'd be surprised of watches are powerful enough. I also assume we can detect if the app runs on a watch and potentially adjust the layout accordingly (specifically for the rounded frame)?

Reply:
> Is the full app running on the watch? Or does it run in some sort of stripped down way? 

I haven't passed the "Connecting..." phase, even the watch is connected to the internet and everything, I'd need to debug and check deeper, but I doubt it, I was just going a bit further with the testing of this PR and since I was going thru the [Android guide](https://github.com/bitcoin-core/gui-qml/pull/366) it seems I got a bit overexcited :laughing:.

> I'd be surprised of watches are powerful enough. 

Well, this is an old one, 2017-ish, nowadays resources on smartwatches are still limited but they are getting there.

> I also assume we can detect if the app runs on a watch and potentially adjust the layout accordingly (specifically for the rounded frame)?

I'd think so, I'll check when I do something related with it.